### PR TITLE
Make redis host configurable in tests

### PR DESCRIPTION
**Pull request by jakzal**
| Q | A |
| --- | --- |
| Branch? | 3.1 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Makes running tests on dockerized environments possible, since we can now export the `REDIS_HOST` environment variable:

```
REDIS_HOST=redis ./phpunit src/Symfony/Component/Cache
```

If this gets merged, in master we can later replace the `SYMFONY__REDIS_HOST` usage in the FrameworkBundle with the new `env(REDIS_HOST)`.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/CachePoolsTest.php [hunk 1]
```diff
@@ -50,7 +60,7 @@ public function testRedisCustomCachePools()
         try {
             $this->doTestCachePools(array('root_config' => 'redis_custom_config.yml', 'environment' => 'custom_redis_cache'), RedisAdapter::class);
         } catch (\PHPUnit_Framework_Error_Warning $e) {
-            if (0 !== strpos($e->getMessage(), 'unable to connect to 127.0.0.1')) {
+            if (0 !== strpos($e->getMessage(), 'unable to connect to')) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/CachePoolsTest.php]: Not sure the behavior is the same with phpredis & predis. This should cover both.

Comment (jakzal) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/CachePoolsTest.php]: Both return false and behaviour will be the same (exception never thrown). Note that this doesn't apply to the previous test, where exception is indeed thrown and the test is skipped properly.

Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/CachePoolsTest.php]: not php warning either generated by phpunit? then feel free to remove the copy/paste :)

Comment (jakzal) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/CachePoolsTest.php]: Nothing :) i don't have a problem with the copy&paste at the moment. My problem is if tests fail to connect to redis they're not skipped. Not sure how to solve it yet, so I'll leave it for now.

**Comment by fabpot** [comment]
Thank you @jakzal.

**Pull request closed without merging**